token-handler/src/mint_tokens.rs

```diff
@@ -100,6 +82,41 @@ pub trait MintTokensModule:
         self.distribute_payments(&hash_of_hashes, &operation_tuple);
     }
 
+    fn prepare_nft_create_args(
+        &self,
+        token_identifier: &TokenIdentifier<Self::Api>,
+        token_nonce: &u64,
+        token_data: &StolenFromFrameworkEsdtTokenData<Self::Api>,
+    ) -> ManagedArgBuffer<Self::Api> {
+        let mut arg_buffer = ManagedArgBuffer::new();
+        let cloned_token_data = token_data.clone();
```

Comment:
@mihaicalinluca do you think there is a more efficient way of using this reference without cloning it ? `push_arg` consumes the parameter

Reply:
from my testing you can just use references there as such:

```
arg_buffer.push_arg(&token_identifier);
        arg_buffer.push_arg(&token_data.amount);
        arg_buffer.push_arg(&token_data.name);
        arg_buffer.push_arg(&token_data.royalties);
        arg_buffer.push_arg(&token_data.hash);
        arg_buffer.push_arg(&token_data.attributes);
```